### PR TITLE
Add error handling for empty DataLoader

### PR DESCRIPTION
**Pull request by perliedman**
## Overview

This adds two small error checks for when the `DataLoader` returns an empty iterator. Without this, code crashes with hard-to-understand internal errors.

### Checklist

- [ ] Added `needs-backport` label if PR is bug fix that applies to previous minor release
- [X] Ran scripts/format_code and committed any changes
- [X] Documentation updated if needed
- [X] PR has a name that won't get you publicly shamed for vagueness

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

Learner:

* Create a dataset which is empty (for example, a random window sampler with an AOI that is impossible to hit)
* Run `train_epoch`
* `ValueError` should be thrown (previosly resulted in an `IndexError` when `outputs` was accessed)

Visualizer:

* Create a dataset which is empty (for example, a random window sampler with an AOI that is impossible to hit)
* Call `plot_batch`
* `ValueError` should be thrown (previously resulted in an `NameError` since `x` wasn't defined)

### CODE REVIEW

**Comment by AdeelH** [comment]
Thank you for the PR! 

This looks good to me, but I'll let @lewfish take a look in case the `for x, y in dl: break` construction was important for some reason.

I think a unit test for `Learner.train_epoch()` might be tricky/too involved, but a unit test for `Visualizer.get_batch()` would be nice.

File: rastervision_pytorch_learner/rastervision/pytorch_learner/dataset/visualizer/visualizer.py
Comment (lewfish) [inline rastervision_pytorch_learner/rastervision/pytorch_learner/dataset/visualizer/visualizer.py:157]: LGTM. I forgot I could just use `next(iter(dl))`

**Comment by lewfish** [comment]
Thanks for the contribution. Even though it's a very small change we'll need for you to sign a CLA before merging. Sorry for the trouble. See https://github.com/azavea/raster-vision#contributing

**Comment by perliedman** [comment]
@lewfish :+1: no worries, I e-mailed you the license agreement just now!

**Review by AdeelH — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**
